data/data/bootstrap/baremetal/files/usr/local/bin/startironic.sh.template

```diff
@@ -0,0 +1,75 @@
+
+# This image is templated in via the installer pkg/asset/ignition/bootstrap/bootstrap.go
+RHCOS_IMAGE_URL="{{.DeployImage}}"
+
+# Start the provisioning nic if not already started
+# FIXME we should detect the nic in the ironic container (currently relies on the ifname, and won't pick 
+# up the description added here)
+# FIXME The IP/subnet and gateway should be provided via survey input?
+# FIXME: the ironic containers currently bind to all interfaces, which is not secure in the
+# case of the cluster, but we'll need the installer to access the public nic on the boostrap node
+# so probably we'll need a forwarding rule after https://github.com/metal3-io/ironic-image/pull/56
+PROVISIONING_NIC=ens4
+if ! nmcli -t device | grep "$PROVISIONING_NIC:ethernet:connected:provisioning"; then
+    nmcli c add type ethernet ifname $PROVISIONING_NIC con-name provisioning ip4 172.22.0.2/24 gw4 172.22.0.1
+fi
+
+# set password for mariadb
+mariadb_password=$(echo $(date;hostname)|sha256sum |cut -c-20)
+
+IRONIC_SHARED_VOLUME="ironic"
+podman volume create $IRONIC_SHARED_VOLUME || true
+
+# Add firewall rules to ensure the IPA ramdisk can reach httpd, Ironic and the Inspector API on the host
+for port in 80 5050 6385 ; do
+    if ! sudo iptables -C INPUT -i $PROVISIONING_NIC -p tcp -m tcp --dport $port -j ACCEPT > /dev/null 2>&1; then
+        sudo iptables -I INPUT -i $PROVISIONING_NIC -p tcp -m tcp --dport $port -j ACCEPT
+    fi
+done
+
+# Start dnsmasq, http, mariadb, and ironic containers using same image
+# FIXME with the current version of podman/libpod on the bootstrap VM this
+# breaks when trying to use a pod to encapsulate the containers
+podman run -d --net host --privileged --name mariadb \
+     -v $IRONIC_SHARED_VOLUME:/shared:z --entrypoint /bin/runmariadb \
+     --env MARIADB_PASSWORD=$mariadb_password ${IRONIC_IMAGE}
+
+podman run -d --net host --privileged --name dnsmasq \
+     --env PROVISIONING_INTERFACE=$PROVISIONING_NIC \
+     -v $IRONIC_SHARED_VOLUME:/shared:z --entrypoint /bin/rundnsmasq ${IRONIC_IMAGE}
+
+podman run -d --net host --privileged --name httpd \
+     --env PROVISIONING_INTERFACE=$PROVISIONING_NIC \
+     -v $IRONIC_SHARED_VOLUME:/shared:z --entrypoint /bin/runhttpd ${IRONIC_IMAGE}
+
+podman run -d --net host --name ipa-downloader \
+     --env CACHEURL="http://172.22.0.1/images" \
+     -v $IRONIC_SHARED_VOLUME:/shared:z ${IPA_DOWNLOADER_IMAGE} /usr/local/bin/get-resource.sh
+
+podman run -d --net host --name coreos-downloader \
+     --env CACHEURL="http://172.22.0.1/images" \
+     -v $IRONIC_SHARED_VOLUME:/shared:z ${COREOS_DOWNLOADER_IMAGE} /usr/local/bin/get-resource.sh $RHCOS_IMAGE_URL
+
+podman run -d --net host --privileged --name ironic-inspector \
+     --env PROVISIONING_INTERFACE=$PROVISIONING_NIC \
+     -v $IRONIC_SHARED_VOLUME:/shared:z "${IRONIC_INSPECTOR_IMAGE}"
+
+podman run -d --net host --privileged --name ironic \
```

Comment:
Let's split this the same way we did it in dev-scripts: https://github.com/openshift-metal3/dev-scripts/pull/646